site/app/controllers/OfficeHoursQueueController.php

```diff
@@ -588,4 +588,12 @@ private function sendSocketMessage(array $msg_array): void {
             $this->core->addNoticeMessage("WebSocket Server is down, page won't load dynamically.");
         }
     }
+
+    /**
+     * @Route("/courses/{_semester}/{_course}/office_hours_queue/preview", methods={"POST"})
+     * @AccessControl(role="LIMITED_ACCESS_GRADER")
+     */
+    public function showMarkdownPreview() {
+        $this->core->getOutput()->renderOutput('OfficeHoursQueue', 'previewAnnouncement', $_POST['enablePreview'], $_POST['content']);
+    }
```

Comment:
Looking at this, to me, these preview controller functions and their related view functions are mostly the same with the only difference being in who can access them. However, given that this is only rendering back content as provided by the user, the access control is unnecessary as we're not hiding anything sensitive.

With that, it would be good to condense all of these into a singular controller method in a new `MarkdownController` and related singular view method in `MarkdownView`.